Faulty `.module` file since version 1.3.0

Running `gradle build` on the following:

```
buildscript {
    repositories {
        repositories {
            maven {
                url 'https://plugins.gradle.org/m2'
                metadataSources {
                    gradleMetadata()
                    mavenPom()
                }
            }
        }

    }
    dependencies {
        classpath "org.gradle:test-retry-gradle-plugin:1.3.0"
    }
}

apply plugin: "org.gradle.test-retry"
```

Causes an error:

```
FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring root project 'gradle_retry_reproducer'.
> Could not resolve all files for configuration ':classpath'.
   > Could not find plugin-1.3.0.jar (org.gradle:test-retry-gradle-plugin:1.3.0).
     Searched in the following locations:
         https://plugins.gradle.org/m2/org/gradle/test-retry-gradle-plugin/1.3.0/plugin-1.3.0.jar
```


This appears to be because of the faulty jar file path `plugin-1.3.0.jar` included in the `.module` file: \
https://plugins.gradle.org/m2/org/gradle/test-retry-gradle-plugin/1.3.0/test-retry-gradle-plugin-1.3.0.module

The jar file path _should be_ `test-retry-gradle-plugin-1.3.0.jar`

# Notes:
- This error _does not_ occur with version 1.2.1 or below, because those versions do not have a `.module` file.
- This error _does_ occur with version 1.3.0 and above, including the latest version 1.5.2, because those versions all have a `.module` file with a faulty `.jar` path.
- This error does not occur when using `gradlePluginPortal()` as a repository, because it does not use `gradleMetadata()`
- Tested with Gradle versions {7.5.1, 8.1}

--- a/.teamcity/settings.kts
+++ b/.teamcity/settings.kts
@@ -138,7 +138,7 @@ project {
 
             steps {
                 gradle {
-                    tasks = "clean snapshot publishPluginPublicationToGradleBuildInternalSnapshotsRepository -x test"
+                    tasks = "clean snapshot publishPluginMavenPublicationToGradleBuildInternalSnapshotsRepository -x test"
                     gradleParams =
                         "-s $useGradleInternalScansServer $buildCacheSetup $useGradleInternalScansServer"
                     param("org.jfrog.artifactory.selectedDeployableServer.defaultModuleVersionConfiguration", "GLOBAL")

--- a/plugin/build.gradle.kts
+++ b/plugin/build.gradle.kts
@@ -1,8 +1,7 @@
-import com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar
+import com.google.gson.Gson
 import org.gradle.testretry.build.GradleVersionData
 import org.gradle.testretry.build.GradleVersionsCommandLineArgumentProvider
 import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
-import com.google.gson.Gson
 import java.net.URL
 
 plugins {
@@ -44,9 +43,14 @@ tasks.withType<KotlinCompile>().configureEach {
     kotlinOptions.jvmTarget = "1.8"
 }
 
-val plugin: Configuration by configurations.creating
+val plugin: Configuration by configurations.creating {
+    isCanBeConsumed = false
+    isCanBeResolved = true
+}
 
-configurations.getByName("compileOnly").extendsFrom(plugin)
+configurations.compileOnly {
+    extendsFrom(plugin)
+}
 
 dependencies {
     val asmVersion = "9.5"
@@ -63,8 +67,7 @@ dependencies {
     codenarc("org.codenarc:CodeNarc:3.2.0")
 }
 
-val shadowJar = tasks.named<ShadowJar>("shadowJar")
-shadowJar.configure {
+tasks.shadowJar {
     configurations = listOf(plugin)
     dependencies {
         include(dependency("org.ow2.asm:asm"))
@@ -78,7 +81,7 @@ shadowJar.configure {
 
 tasks.jar {
     enabled = false
-    dependsOn(shadowJar)
+    dependsOn(tasks.shadowJar)
 }
 
 gradlePlugin {
@@ -140,26 +143,8 @@ publishing {
     }
 }
 
-configurations {
-    configureEach {
-        outgoing {
-            val removed = artifacts.removeIf { it.classifier.isNullOrEmpty() }
-            if (removed) {
-                artifact(tasks.shadowJar) {
-                    classifier = ""
-                }
-            }
-        }
-    }
-    // used by plugin-publish plugin
-    archives {
-        extendsFrom(signatures.get())
-    }
-}
-
 signing {
     useInMemoryPgpKeys(System.getenv("PGP_SIGNING_KEY"), System.getenv("PGP_SIGNING_KEY_PASSPHRASE"))
-    sign(configurations.archives.get())
 }
 
 tasks.withType<Sign>().configureEach {

--- a/plugin/build.gradle.kts
+++ b/plugin/build.gradle.kts
@@ -14,7 +14,7 @@ plugins {
     codenarc
     `kotlin-dsl`
     signing
-    id("com.gradle.plugin-publish") version "0.15.0"
+    id("com.gradle.plugin-publish") version "1.2.0"
     id("com.github.hierynomus.license") version "0.16.1"
     id("com.github.johnrengelman.shadow") version "8.1.1"
 }
@@ -82,30 +82,21 @@ tasks.jar {
 }
 
 gradlePlugin {
+    website.set("https://github.com/gradle/test-retry-gradle-plugin")
+    vcsUrl.set("https://github.com/gradle/test-retry-gradle-plugin.git")
     plugins {
         register("testRetry") {
             id = "org.gradle.test-retry"
             displayName = "Gradle test retry plugin"
             description = project.description
             implementationClass = "org.gradle.testretry.TestRetryPlugin"
+            tags.addAll("test", "flaky")
         }
     }
 }
 
 tasks.pluginUnderTestMetadata {
     pluginClasspath.from(plugin)
-}
-
-pluginBundle {
-    website = "https://github.com/gradle/test-retry-gradle-plugin"
-    vcsUrl = "https://github.com/gradle/test-retry-gradle-plugin.git"
-    description = project.description
-    tags = listOf("test", "flaky")
-
-    mavenCoordinates {
-        groupId = "org.gradle"
-        artifactId = "test-retry-gradle-plugin"
-    }
 }
 
 license {
@@ -124,11 +115,9 @@ license {
 
 publishing {
     publications {
-        register<MavenPublication>("plugin") {
+        create<MavenPublication>("pluginMaven") {
+            groupId = "org.gradle"
             artifactId = "test-retry-gradle-plugin"
-            artifact(shadowJar.get()) {
-                classifier = null
-            }
             pom {
                 licenses {
                     license {

